### DISCUSSION TITLE
How is Avalonia.Android going to deal w/ no more Xamarin?

### DISCUSSION THREAD

**Discussion by tkefauver** [AvaloniaUI/Avalonia#17394] [Q&A] [answered]
Since Xamarin is defunct now what is going to happen w/ Avalonia.Android? 

I noticed it uses Xamarin.AndroidX.AppCompat which is still up-to-date but what's going to happen in the future?

**Comment by maxkatz6** [comment]
Xamarin as a company is defunct.
Xamarin as a series of products owned now by Microsoft is still alive.  

> I noticed it uses Xamarin.AndroidX.AppCompat which is still up-to-date but what's going to happen in the future?

I don't think it's going anywhere in the near future https://github.com/xamarin/AndroidX

**Comment by Coloryr** [comment]
I don't understand the relationship between the `Xamarin.Android` and `Xamarin.AndroidX`
`Xamarin.AndroidX` still update?

**Reply by thevortexcloud** [reply, marked as answer]
Despite what Microsoft says, Xamarin (and Mono) is not dead. They basically just renamed/rebranded/shuffled around most of it. For example, any of the mobile specific TFM's (EG `net6.0-android`) implicitly include Xamarin dependencies.

https://learn.microsoft.com/en-us/dotnet/standard/frameworks

This is the official statement on it:

> Xamarin.Android, Xamarin.iOS, Xamarin.Mac are now integrated directly into .NET (starting with .NET 6) as .NET for Android, .NET for iOS, and .NET for Mac. If you're building with these project types today, they should be upgraded to .NET SDK-style projects for continued support.
>
> Xamarin.Forms has evolved into .NET Multi-platform App UI (MAUI) and existing Xamarin.Forms projects should be migrated to .NET MAUI.
>
> Xamarin support ended on May 1, 2024 for all Xamarin SDKs including Xamarin.Forms. Android API 34 and Xcode 15 SDKs (iOS and iPadOS 17, macOS 14) are the final versions Xamarin targets from existing Xamarin SDKs (that is, no new APIs are planned).

https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin

Microsoft does this with most of their products and keeping track of it all can be very hard. The amount of unnecessary rebranding they do is way too high.

**Reply by tkefauver** [reply]
> keeping track of it all can be very hard

It IS confusing!

**Reply by thevortexcloud** [reply]
Yeah, I still regularly call a lot of Microsoft products by their old names out of habit. EG Office 365, instead of Microsoft 365.

But in the case of Xamarin, there is no way they will ever fully drop it. Xamarin (and Mono) are the only functioning implementations of .NET for android and iOS. They would have to rewrite it all from scratch if they wanted to dump it. Which is why they have instead actually been updating Mono to support all the new .NET features rather than porting the modern version of .NET to those platforms. 

In the case of Mono though, they have taken the interesting step of basically donating the original code to the Wine team, and maintaining a fork of it.

https://www.mono-project.com/
